### DISCUSSION TITLE
Is it possible to deploy this project to AWS directly?

### DISCUSSION THREAD

**Discussion by whoisbigo** [UnlyEd/next-right-now#257] [Q&A] [answered]
Hello,
I'm new to web development.
I'm trying to deploy this project directly to AWS with serverless framework(https://github.com/serverless-nextjs/serverless-next.js).
But after deploying, error message 'Unexpected "serializedDataset" passed as page props.' is showing in browser.
Here is url - https://d2x6auhl6gozrz.cloudfront.net/

*Update
I found that this problem happens to new pages that I created.
Original pages are showing well.

Please give me a hint about this error.
Thank you.

**Comment by Vadorequest** [comment, marked as answer]
This project is meant to be deployed on Vercel using the serverless mode. Why do you want to deploy it to AWS yourself? 

> **That's what Vercel does for you**, but it does it better by using both AWS Lambda and some CDN edge and lots of stuff.

We won't cover/help with issues regarding deployment to other providers, such as deploying to AWS directly. We don't recommend it, there is absolutely no technical benefit to do so yourself.

The `serializedDataset` is expected by Next Right Now as part of the demo, it contains the dataset containing all the dynamic page data. See https://github.com/UnlyEd/next-right-now/blob/5ad05e8cee1caeecfd21f535c136ca1bba500929/src/utils/nextjs/SSG.ts#L181

Returning an empty object as string would work, if you wish not to use it.

**Reply by Vadorequest** [reply]
Actually, returning an empty object won't work either. 

That's because the NRN base code expects your site to belong to a "Customer", which is described in:
https://github.com/UnlyEd/next-right-now/blob/5a04c14f84756646568f7f4fec98b73964cc4c5c/src/types/data/Customer.ts#L17-L32

The serialized dataset is provided from the `getStaticProps` or `getServerSideProps` function of your page and is analyzed by the `MultiversalAppBootstrap` (very similar to `_app`). See below:

https://github.com/UnlyEd/next-right-now/blob/b78b03296b9977ef2d9550418219dac0326fbfb1/src/components/appBootstrap/MultiversalAppBootstrap.tsx#L162

https://github.com/UnlyEd/next-right-now/blob/b78b03296b9977ef2d9550418219dac0326fbfb1/src/components/appBootstrap/MultiversalAppBootstrap.tsx#L214

You should provide a Customer entity for the site to work properly, because it contains the theming configuration, languages which are requirements for any website. 

Depending on your preset, the Demo uses either an Airtable API or a GraphCMS api to fetch those data. 
If you forked this boilerplate, you should first make it work without changing the code (setup your own Airtable base or GraphCMS project and configure it as described in the doc). Then, you can safely change the code and see if you break anything, but it's much easier to start changing stuff from a working setup (and would give you more confidence in the boilerplate).

**Reply by whoisbigo** [reply]
Your answer helps me a lot. Thank you!

**Reply by Vadorequest** [reply]
I'm glad it does! I'll mark this as answered then :)

**Comment by zajcomm** [comment]
This is the same question i have as we are also considering AWS rather than Vercel (for AWS services and HIPAA compliance reasons). Please let me know if you found an answer.

**Reply by Vadorequest** [reply]
Have a look at this: https://dev.to/slsbytheodo/deploy-a-nextjs-app-for-free-on-aws-with-sst-3g28
